repair/repair.cc

```diff
@@ -2253,6 +2253,45 @@ future<> repair::tablet_repair_task_impl::run() {
         auto start_time = std::chrono::steady_clock::now();
         auto parent_data = get_repair_uniq_id().task_info;
         std::atomic<int> idx{1};
+
+        // Start the off strategy updater
+        std::unordered_set<gms::inet_address> participants;
+        std::unordered_set<table_id> table_ids;
+        for (auto& meta : _metas) {
+            thread::maybe_yield();
+            participants.insert(meta.neighbors.all.begin(), meta.neighbors.all.end());
+            table_ids.insert(meta.tid);
+        }
+        abort_source as;
+        auto off_strategy_updater = seastar::async([&rs, uuid = id.uuid().uuid(), &table_ids, &participants, &as] {
+            auto tables = std::list<table_id>(table_ids.begin(), table_ids.end());
+            auto req = node_ops_cmd_request(node_ops_cmd::repair_updater, node_ops_id{uuid}, {}, {}, {}, {}, std::move(tables));
+            auto update_interval = std::chrono::seconds(30);
+            while (!as.abort_requested()) {
+                sleep_abortable(update_interval, as).get();
+                parallel_for_each(participants, [&rs, uuid, &req] (gms::inet_address node) {
+                    return rs._messaging.send_node_ops_cmd(netw::msg_addr(node), req).then([uuid, node] (node_ops_cmd_response resp) {
```

Comment:
This delays off-strategy compaction for all tablets in the table.
As a short-term workaround, that can be tolerated but it doesn't take advantage of the incremental properties of tablet repair that can be triggered immediately after the tablet token range is repaired, or it can be avoided in many cases where offstrategy compaction for tablet-repair is just unneeded.

Reply:
Ideally, the off strategy compaction manager could make decision to trigger the offstrategy even if there is such update which prevents the timer to expire and trigger the offstrategy, in case like there are too many files in the maintenance  set.

Reply:
Just with tablets the maintenance sstable sets are per-tablet and are small, so I don't see it triggering. And in the meanwhile thise sstables are fenced off from regular compaction.
@raphaelsc, how about split compaction?
Do we split sstables in the maintenance set or do we wait for reshape?

Reply:
split as part of tablet splitting.

Reply:
so if we don't split maintenance sstables until offstrategy gets to them, this will delay the distributed tablet split accordingly.

Reply:
> so if we don't split maintenance sstables until after offstrategy gets to them and moves them to the main sstable set, this will delay the distributed tablet split accordingly.

you're right, we need to make sure split also works on maintenance set, to avoid the delay.

Reply:
if we're still writing new sstables to maintenace set though (e.g. repair is still in progress), tablet split will be delayed anyway. split execution phase is serialized with topology changes / repair. so splitting maintenance set only helps after repair activity ceased, but off-strategy haven't started yet.